### PR TITLE
[MGDOBR-1335] Removed sensitive fields from ManagedBridge CRD

### PR DESCRIPTION
**Pull request by robpblake**
- Removed sensitive fields from the ManagedBridge CRD and ensured that they are only written to the Secret associated with the Bridge
- Updated the ManagedBridge CRD for our kustomize overlays
- Tested the ManagedBridge converter which was originally incorrectly mapping TLS key and certificate
- Added labels to all resources created by ManagedBridgeService that allow us to easily locate them using kubectl

[MGDOBR-1335](https://issues.redhat.com/browse/MGDOBR-1335)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-XXX] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Review by manstis — APPROVED** [review]
LGTM :+1:

**Review by manstis — APPROVED** [review]
Approving again; following rebase.

File: shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/ManagedBridgeServiceImpl.java [hunk 1]
```diff
@@ -226,6 +245,8 @@ public AuthorizationPolicy fetchOrCreateBridgeAuthorizationPolicy(ManagedBridge
         expected.getSpec().getSelector().setMatchLabels(Collections.singletonMap(Constants.BRIDGE_INGRESS_AUTHORIZATION_POLICY_SELECTOR_LABEL,
                 istioGatewayProvider.getIstioGatewayService().getMetadata().getLabels().get(Constants.BRIDGE_INGRESS_AUTHORIZATION_POLICY_SELECTOR_LABEL)));
 
+        ensureManagedBridgeLabels(managedBridge, expected);
```
Comment (r00ta) [inline shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/ManagedBridgeServiceImpl.java:248]: Nitpick: move this call to line 233 close to where the metadata is set for the `expected` resource? Same for all the other calls to `ensureManagedBridgeLabels` in the other methods

**Review by r00ta — APPROVED** [review]

File: shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/ManagedBridgeServiceImpl.java (hunk 1 above)
Comment (robpblake) [reply shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/ManagedBridgeServiceImpl.java]: Done 👍

**Pull request merged**